### PR TITLE
apacheHttpd: 2.4.52 -> 2.4.53

### PR DESCRIPTION
**Pull request by mweinelt**
###### Description of changes
https://downloads.apache.org/httpd/CHANGES_2.4.53

The configure failed to find a working pcre-config, so I decided to try
out pcre2 instead, which resulted in the same problem. Putting it into
PATH during preConfigure worked, but it isn't a great workaround.

Migrating to pcre2 was recommended in the release notes, since pcre 8.x
is over 20 years old and has now reached its end of life.

Fixes: CVE-2022-23943, CVE-2022-22721, CVE-2022-22720, CVE-2022-22719

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/http/apache-httpd/2.4.nix [hunk 1]
```diff
@@ -38,11 +38,16 @@ stdenv.mkDerivation rec {
   # Required for ‘pthread_cancel’.
   NIX_LDFLAGS = lib.optionalString (!stdenv.isDarwin) "-lgcc_s";
 
+  # TODO: the configure script fails to find a working pcre2-config without this
+  preConfigure = ''
+    export PATH=${lib.getDev pcre2}/bin:$PATH
+  '';
```
Comment (mweinelt) [inline pkgs/servers/http/apache-httpd/2.4.nix:44]: This is not ideal and if someone has time to check the upstream changes that resulted in that breakage I'm all for it.

Comment (jojosch) [reply pkgs/servers/http/apache-httpd/2.4.nix:44]: The `which` command is missing while trying to determine the pcre2-config location when setting it via [`--with-pcre`](https://github.com/apache/httpd/blob/2.4.53/configure.in#L215-L225). 

Adding `which` to the inputs and setting `--with-prce` to `${pcre2.dev}/bin/pcre2-config` fixes the build without the workaround.

```
...
apache-httpd> checking for gcc option to accept ISO C99... none needed
apache-httpd> /nix/store/3vygpb5d5n7gq5f7k30zvpv2sgnw5r1v-pcre2-10.39-dev/bin/pcre2-config
apache-httpd> checking for -pcre2-config... no
apache-httpd> checking for -pcre-config... no
apache-httpd> checking for pcre2-config... no
apache-httpd> checking for pcre-config... no
apache-httpd> configure: Using external PCRE library from /nix/store/3vygpb5d5n7gq5f7k30zvpv2sgnw5r1v-pcre2-10.39-dev/bin/pcre2-config
apache-httpd>   setting PCRE_INCLUDES to "-I/nix/store/3vygpb5d5n7gq5f7k30zvpv2sgnw5r1v-pcre2-10.39-dev/include"
apache-httpd>   setting PCRE_LIBS to "-L/nix/store/4vxdmr77q3kzf223vzyxbr82mj82203y-pcre2-10.39/lib -lpcre2-8"
apache-httpd> configure:
apache-httpd> configure: Configuring Apache httpd...
...
```

Comment (jojosch) [reply pkgs/servers/http/apache-httpd/2.4.nix:44]: In 2.4.52 the [check](https://github.com/apache/httpd/blob/2.4.52/configure.in#L215-L223) worked without `which` and appended `/bin/pcre-config`to the `--with-pcre` path.

Comment (mweinelt) [reply pkgs/servers/http/apache-httpd/2.4.nix]: Thanks for looking into this.

**Comment by mweinelt** [comment]
@ofborg eval

**Pull request merged**